components/scream/src/share/scream_types.hpp

```diff
@@ -1,18 +1,63 @@
-#ifndef INCLUDE_SCREAM_TYPES
-#define INCLUDE_SCREAM_TYPES
+#ifndef SCREAM_TYPES_HPP
+#define SCREAM_TYPES_HPP
+
+#include <Kokkos_Core.hpp>
 
 #include "share/scream_config.hpp"
 
 namespace scream {
 
 #ifdef SCREAM_DOUBLE_PRECISION
-typedef double Real;
+using Real = double;
 #else
-typedef float Real;
+using Real = float;
 #endif
 
 typedef int Int;
 
+// Detect ExecSpace using some ifdef logic. I put Serial here as a placeholder
+using ExecSpace = Kokkos::Serial;
+using HostSpace = Kokkos::HostSpace;
+
+// Memory spaces
+using ExecMemSpace = ExecSpace::memory_space;
+using HostMemSpace = HostSpace::memory_space;
+
+// Memory traits
+using MemoryManaged   = Kokkos::MemoryTraits<Kokkos::Restrict>;
+using MemoryUnmanaged = Kokkos::MemoryTraits<Kokkos::Restrict | Kokkos::Unmanaged>;
+
+// A general view in Scream has layout right ('row-major'-like)
+template<typename DataType, typename... Properties>
+using ViewType = Kokkos::View<DataType, Kokkos::LayoutRight, Properties...>;
+
+// ======== Some particular view's aliases ========== //
+
+// Managed/Unmanaged view
+template <typename DataType, typename... Properties>
+using ViewManaged = ViewType<DataType, Properties..., MemoryManaged>;
```

Comment:
scream_kokkos_meta.hpp provides an Unmanaged decorator that is more general than this in that it permits other memory traits.

Reply:
(By the way, is it OK to merge the trivial PR #31 to clean up this decorator?)

Reply:
Well, that one is a decorator, a type dependent on another type. If you use unmanged to view another view, then the decorator is perfect. But there will be places (e.g., surface coupling) where we will create unmanaged views from f90 pointers. In those areas, we won't have a reference managed view to deduce the unamanged templates arguments from.

But yes, in the FR I should probably use the unmanaged decorator to get an unmanaged view from a managed one.